assets/scripts/app/MessageBar.jsx

```diff
@@ -67,14 +69,10 @@ class MessageBar extends React.Component {
               {linkText || <FormattedMessage id="msg.more-info" defaultMessage="More info" />}
             </a>
           }
-
-          <button
-            className="close"
+          <CloseButton
             onClick={this.onClickDismiss}
             title={this.props.intl.formatMessage({ id: 'btn.dismiss', defaultMessage: 'Dismiss' })}
```

Comment:
Let's move this title into the `CloseButton` component. All `CloseButton` components should, by default, use the "Dismiss" as a title attribute _unless_ overridden by a prop, e.g. in `GalleryStreetItem` where "Delete street" is the title.

This would allow us to remove the `injectIntl` HOC from this component because we no longer need it here (and possibly other components as well).

Reply:
Does that mean doing away with the `intl` prop? Just passing Ordinary string to the title like

```js
<CloseButton title="Delete street" />
```

Reply:
No - sorry, I wasn't clear.

We should _move_ the `intl` prop into `CloseButton` so that it can display "Dismiss" by default. Components that already specify this:

```
title={this.props.intl.formatMessage({ id: 'btn.dismiss', defaultMessage: 'Dismiss' })}
```

_or_ specify no title will have "Dismiss" by default. This is because the close buttons that have no title _should_ have _a_ title, in order to make them more accessible.

For other components that have different titles, they will need to pass in a string created by `intl`.

Reply:
In that case, how do we test for title? when we have to pass `this.props.intl.formatMessage({})` to title. Also setting the default props like below will not work. 
```js
  static defaultProps = {
    className: 'close',
    title: this.props.intl.formatMessage({
      id: 'btn.dismiss',
      defaultMessage: 'Dismiss'
    })
  }
```

Reply:
That's a great point. In that case it won't be part of the static `defaultProps` property but it will be the fallback value for `title` that must be determined at runtime, in the `render()` function.

Reply:
That will work but how do we test for when `title` prop?
```js
const wrapper = shallow(<CloseButton
      onClick={onClick}
      title={mockIntl.formatMessage({ id: 'gallery.delete-street-tooltip', defaultMessage: 'Delete street' })}
      className="hide-btn"
    />)
```
Running test for the above will require wrapping with `<IntlProvider>`

Reply:
We have a helper that wraps `shallow` with a mock `IntlProvider`, for example:

https://github.com/streetmix/streetmix/blob/bad606c2b7d5dc9e7788f7714b0d68e48d7ea640/assets/scripts/streets/__tests__/StreetMetaWidth.test.js#L4

More info on the react-intl wiki: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#enzyme

You can then get the instance of the mounted element to see if the title attribute is set, for example: https://github.com/streetmix/streetmix/blob/8808e02ea2e619997da574958d316066ea98cbc1/assets/scripts/streets/__tests__/SearchAddress.test.js#L79

Reply:
```js
 expect(el.instance().getAttribute('title')).toEqual('Clear search') 
```
Above does not work using `shallowWithIntl` while `mountWithIntl` works but only the default title is received even when I comment out the default, `null` was received. Although I'm still trying to fig out why.

Reply:
Let me know if you'd like me to take a look.

Reply:
Not really -- what are your thoughts around trying to avoid doing this?

Reply:
```js 
it('should set title attribute', () => {
    const wrapper = mountWithIntl(<CloseButton
      onClick={onClick}
      title={mockIntl.formatMessage({ id: 'gallery.delete-street-tooltip', defaultMessage: 'Delete street' })}
    />)
    expect(wrapper.find('button').instance().getAttribute('title')).toEqual('Delete street')
  })
``` 
Above is the test I wrote to check for the attribute. It fails because the `title` attribute is still set to default (`"Dismiss"`).

Reply:
@flickz OK I see what's up. You're working off an older branch, where our `mockIntl.formatMessage()` is just a no-op. This returns nothing, which is why the title is defaulting to `Dismiss`. If you update your branch, you shold get an updated version of `mockIntl.formatMessage()` which returns the `defaultMessage` automatically, which should make your test pass.

Reply:
Oh yeah. Works now! Thanks. I will close this and send a new PR from the branch